src/pylutron_caseta/smartbridge.py

```diff
@@ -251,14 +251,32 @@ async def _request(
         if self._leap is None:
             raise BridgeDisconnectedError()
 
-        async with asyncio_timeout(REQUEST_TIMEOUT):
-            response = await self._leap.request(communique_type, url, body)
+        # LEAP APIs support pagination, so repeat requests until fully collected
+        responses = []
+        paging = None
 
-        status = response.Header.StatusCode
-        if status is None or not status.is_successful():
-            raise BridgeResponseError(response)
+        while True:
+            async with asyncio_timeout(REQUEST_TIMEOUT):
+                response = await self._leap.request(communique_type, url, body, paging=paging)
+
+            status = response.Header.StatusCode
+            if status is None or not status.is_successful():
+                raise BridgeResponseError(response)
+
+            responses.append(response)
+
+            paging = response.Header.Paging
+            if not paging:
+                break
+
+        # merge the Body of multiple paged Responses together
+        merged = responses.pop(0)
+        if merged.Body:
+            merged_type = list(merged.Body.keys())[0]
```

Comment:
I don't know if this is quite right. Maybe we should merge every value which contains a list. The problem with taking the first one is that the order of JSON properties usually has no meaning.

Reply:
Yeah, hard to know without authoritative protocol documentation. I'm not sure if there are responses that contain multiple entity types within the Body, but merging all the lists with the same top-level entity type seems reasonable?